learning_resources_search/models.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.db.models import JSONField
+
+from main.models import TimestampedModel
+
+User = get_user_model()
+
+
+class PercolateQuery(TimestampedModel):
+    """An opensearch query used in percolate"""
+
+    SEARCH_SUBSCRIPTION_TYPE = "search_subscription_type"
+
+    SOURCE_TYPES = [
+        SEARCH_SUBSCRIPTION_TYPE,
+    ]
+
+    original_query = JSONField()
+    query = JSONField()
+    source_type = models.CharField(
+        max_length=255, choices=[(choice, choice) for choice in SOURCE_TYPES]
+    )
+
+    def __str__(self):
+        return f"Percolate query {self.id}: {self.query}"
+
+
+class PercolateQueryUser(TimestampedModel):
+    """
```

Comment:
What do you think of adding a `ManyToMany` relationship to `User` on the `PercolatorQuery` model instead of creating this model?  I think the end result would be a similar table in the database.

```python
users = ManyToManyField(
    User, 
    related_name="percolate_queries"
). # many users possibly associated to query, many queries possibly associated to user
```

Reply:
The reason behind that intermediate model is to have something that stores the reason for subscribing to the query. so for example, in future use cases we might have a "subscribe to physics" newletters in which case we would have a broad query stored "topic=physics" - some users may have saved this via site-search and subscribed to it via the button and other users are subscribed to it through some newsletter campaign.